data_structures/trie/trie.py

```diff
@@ -46,6 +46,57 @@ def find(self, word: str) -> bool:
             curr = curr.nodes[char]
         return curr.is_leaf
 
+    def match(self, prefix: str) -> [str]:
```

Comment:
`match()` seems like a weak function name.  What about `words_starting_with()` or `get_words_starting_with()`?

Doctests?  How do we know this works?